lib/console.js

```diff
@@ -9,25 +9,24 @@ function Console(stdout, stderr) {
   if (!stdout || typeof stdout.write !== 'function') {
     throw new TypeError('Console expects a writable stream instance');
   }
-  if (!stderr) {
-    stderr = stdout;
-  }
-  var prop = {
+  const prop = {
     writable: true,
     enumerable: false,
-    configurable: true
+    configurable: true,
+    value: stdout
   };
-  prop.value = stdout;
   Object.defineProperty(this, '_stdout', prop);
-  prop.value = stderr;
+  if (stderr) {
+    prop.value = stderr;
+  }
   Object.defineProperty(this, '_stderr', prop);
   prop.value = new Map();
   Object.defineProperty(this, '_times', prop);
 
   // bind the prototype functions to this Console instance
-  var keys = Object.keys(Console.prototype);
-  for (var v = 0; v < keys.length; v++) {
-    var k = keys[v];
+  const keys = Object.keys(Console.prototype);
+  for (let v = 0; v < keys.length; v++) {
```

Comment:
If anyone is interested in results. Seems, caching of array length does not help at all.

[irhydra-array-length.zip](https://github.com/nodejs/node/files/108279/irhydra-array-length.zip)